core/templates/domain/topic/editable-topic-backend-api.service.ts

```diff
@@ -16,233 +16,298 @@
-          topic_name: topicName
+    let putData = {
+      version: topicVersion,
+      commit_message: commitMessage,
+      topic_and_subtopic_page_change_dicts: changeList
+    };
+    this.http.put<UpdateTopicBackendResponse>(
+      editableTopicDataUrl, putData).toPromise().then((response) => {
+      if (successCallback) {
+        // Here also, a dict with 2 fields are passed instead of just 2
+        // parameters, due to the same reason as written for _fetchTopic().
+        successCallback({
+          topicDict: response.topic_dict,
+          skillIdToDescriptionDict: response.skill_id_to_description_dict,
+          skillIdToRubricsDict: response.skill_id_to_rubrics_dict
         });
-      $http.get(topicNameUrl).then(function(response) {
-        if (successCallback) {
-          successCallback(response.data.topic_name_exists);
-        }
-      }, function(errorResponse) {
-        if (errorCallback) {
-          errorCallback(errorResponse.data);
-        }
+      }
+    }, (errorResponse) => {
+      if (errorCallback) {
+        errorCallback(errorResponse.error);
+      }
+    });
+  }
+
+  private _doesTopicWithUrlFragmentExist(
+      topicUrlFragment: string,
+      successCallback: (value?: Object | PromiseLike<Object>) => void,
+      errorCallback: (reason?: string) => void): void {
+    let topicUrlFragmentUrl = this.urlInterpolationService.interpolateUrl(
+      TopicDomainConstants.TOPIC_URL_FRAGMENT_HANDLER_URL_TEMPLATE, {
+        topic_url_fragment: topicUrlFragment
       });
-    };
+    this.http.get<DoesTopicWithUrlFragmentExistBackendResponse>(
+      topicUrlFragmentUrl).toPromise().then((response) => {
+      if (successCallback) {
+        successCallback(response.topic_url_fragment_exists);
+      }
+    }, (errorResponse) => {
+      if (errorCallback) {
+        errorCallback(errorResponse.error);
+      }
+    });
+  }
 
-    return {
-      fetchTopic: function(topicId) {
-        return $q(function(resolve, reject) {
-          _fetchTopic(topicId, resolve, reject);
-        });
-      },
+  private _doesTopicWithNameExist(
+      topicName: string,
+      successCallback: (value?: Object | PromiseLike<Object>) => void,
+      errorCallback: (reason?: string) => void): void {
+    let topicNameUrl = this.urlInterpolationService.interpolateUrl(
+      TopicDomainConstants.TOPIC_NAME_HANDLER_URL_TEMPLATE, {
+        topic_name: topicName
+      });
+    this.http.get<DoesTopicWithNameExistBackendResponse>(
+      topicNameUrl).toPromise().then((response) => {
+      if (successCallback) {
+        successCallback(response.topic_name_exists);
+      }
+    }, (errorResponse) => {
+      if (errorCallback) {
+        errorCallback(errorResponse.error);
+      }
+    });
+  }
 
-      fetchStories: function(topicId) {
-        return $q(function(resolve, reject) {
-          _fetchStories(topicId, resolve, reject);
-        });
-      },
+  fetchTopic(topicId: string): Promise<Object> {
```

Comment:
Please change 'Object' to the appropriate type.

Reply:
Done